trade-finance-manager-ui/server/controllers/case/parties/helpers.js

```diff
@@ -3,4 +3,50 @@ const { userIsInTeam } = require('../../../helpers/user');
 
 const userCanEdit = (user) => userIsInTeam(user, [CONSTANTS.TEAMS.BUSINESS_SUPPORT]);
 
-module.exports = userCanEdit;
+// takes body if bondIssuer or bond beneficiary and returns body and party type
+const bondPartyType = (body) => {
+  if (body.bondBeneficiaryPartyUrn) {
+    return {
+      bodyType: 'bondBeneficiaryPartyUrn',
+      partyType: 'bonds-beneficiary',
+    };
+  }
+
+  return {
+    bodyType: 'bondIssuerPartyUrn',
+    partyType: 'bonds-issuer',
+  };
+};
+
+// checks if bond type and returns true or false
+const isBondPartyType = (partyType) => (partyType === 'bondBeneficiaryPartyUrn' || partyType === 'bondIssuerPartyUrn');
+
+/**
+ * constructs errRef based on partyType
+ * if bondBeneficiary or bondIssuer, then needs index to specify which urn box it is referring to,
+ * else should just be partyUrn
+ */
+const constructErrRef = (partyType, index) => {
+  let errRef = 'partyUrn';
+
+  if (isBondPartyType(partyType)) {
+    errRef = `partyUrn-${index}`;
+  }
+
+  return errRef;
+};
+
+const isEmptyString = (str) => {
+  if (!str || ((typeof str === 'string' || str instanceof String) && !str.trim().length)) {
```

Comment:
Do we need to make `typeof` and `instaceof` condition an `&&` condition. As you can pass an `undefined` object which will return true.

Is the function only expecting String typeof?

Reply:
discussed, leaving as is as will never be undefined or null and validation can handle

Reply:
discussed, leaving as is as will never be undefined or null and validation can handle